### ISSUE TITLE
spawned task in scope is not auto cancelled

### ISSUE THREAD

**Issue by nanne007** [pandaman64/task_scope#1]
First, thanks for the great work!
I have a doubt on _cancel spawned task in scope_.
The test case demonstrated in the repo,

``` rust
#[tokio::test]
async fn test_drop() {
    timeout(
        Duration::from_millis(50),
        scope(async {
            spawn(async {
                println!("child started");
                delay_for(Duration::from_millis(100)).await;

                panic!("child is canceled");
            })
            .await;
        }),
    )
    .await
    .unwrap_err();
}
```

shows that the spawned child task should be cancelled after 50ms timeout.

However if we sleep a little bit more after timeout, we can see the `panic!`

``` rust
#[tokio::test]
async fn test_drop_not_auto_cancel() {
    timeout(
        Duration::from_millis(50),
        scope(async {
            spawn(async {
                println!("child started");
                delay_for(Duration::from_millis(100)).await;

                panic!("child is canceled");
            })
            .await;
        }),
    )
    .await
    .unwrap_err();
    println!("timeout-ed");
    delay_for(Duration::from_millis(60)).await;
}
```

I believe the origin test can pass because runtime drops the child task directly, not because it is cancelled due to scope drop.

**Comment by nanne007** [comment]
wrap the child task with `cancelable` will make it work:

``` rust
#[tokio::test]
async fn test_drop_auto_cancel() {
    timeout(
        Duration::from_millis(50),
        scope(async {
            spawn(cancelable(async {
                println!("child started");
                delay_for(Duration::from_millis(100)).await;

                panic!("child should be canceled");
            }))
            .await;
        }),
    )
    .await
    .unwrap_err();
    println!("timeout-ed");
    delay_for(Duration::from_millis(60)).await;
}
```

Or, we can cancel the spawned child task in `spawn` function without wrap in a `cancelable` latter.

**Comment by pandaman64** [comment]
Thank you for the report!
It looks like [spawned future checks cancellation only at the first poll, not running the inter future](https://github.com/pandaman64/task_scope/blob/4639311604a075c346a0f13a2628d2a3d5a0f1c9/src/spawn.rs#L48)😕.
I'll publish a fix in a day.

**Comment by nanne007** [comment]
Yes, great! :)

**Comment by pandaman64** [comment]
The bug should be fixed in `0.1.1`. Thanks!
